cmd/crossplane/main.go

```diff
@@ -84,7 +94,7 @@ func main() {
 		runtimelog.SetLogger(zl)
 	}
 
-	var setupWithManagerFunc func(manager.Manager) error
+	var setupWithManagerFunc func(manager.Manager, appConfig) error
```

Comment:
This doesn't need to block this PR, but I feel like this `setupWithManagerFunc` pattern is getting a little awkward, especially now that we're adding an `appconfig` struct that is only used by one of the two variants.

Reply:
I refactored a bit which seems cleaner.